### DISCUSSION TITLE
CSR instructions not  not recognized

### DISCUSSION THREAD

**Discussion by n-kazdaghli** [xpack-dev-tools/riscv-none-elf-gcc-xpack#20] [Q&A]
Hello,
I am using the RISC-V (riscv-none-elf-gcc-xpack) v12.2.0-3 toolchain, and I cannot access CSR (Control and Status Register) registers. Here is the message I received when I tried to read the MISA (Machine Information Status Register A) registers:
 'unrecognized opcode `csrr a5, misa'."

Is CSR instructions supported by the toolchain?
if not, is there any way to access CSR registers?

**Comment by ilg-ul** [comment]
> Is CSR instructions supported by the toolchain?

CSR instructions are supported.

>  is there any way to access CSR registers?

Yes, invoke the compiler with the proper extensions.

- https://xpack.github.io/blog/2023/02/05/riscv-none-elf-gcc-v12-2-0-3-released/#risc-v-isa-updates
